draft-ietf-httpbis-priority.md

```diff
@@ -501,8 +501,68 @@ not specify the incremental(`i`) parameter.
 
 # Client Scheduling
 
-A client MAY use priority values to make local scheduling choices about the
-requests it initiates.
+A client MAY use priority values to make local processing or scheduling choices
+about the requests it initiates.
+
+
+# Server Scheduling
+
+Priority signals are input to a prioritization process. They do not guarantee
+any particular processing or transmission order for one response relative to any
+other response. An endpoint cannot force a peer to process concurrent request in
+a particular order using priority. Expressing priority is therefore only a
+suggestion.
+
+A server can use priority signals along with other inputs to make scheduling
+decisions. No guidance is provided about how this can or should be done. Factors
+such as implementation choices or deployment environment also play a role. Any
+given connection is likely to have many dynamic permutations. For these reasons,
+there is no unilateral perfect scheduler and this document only provides some
+basic recommendations for implementations. 
+
+Clients cannot depend on particular treatment based on priority signals. Servers
+can use other information to prioritize responses.
+
+It is RECOMMENDED that, when possible, servers respect urgency values. Sending
+higher urgency responses before lower urgency responses.
+
+It is RECOMMENDED that, when possible, servers respect incremental values.
+Sending non-incremental responses of the same urgency one-by-one, and sending
+incremental responses of the same urgency in round-robin manner.
+
+It is RECOMMENDED that a server considers request generation order as an input
+to prioritization. Prioritizing concurrent requests at the same urgency level
+based on the Stream ID, which corresponds to the order in which clients make
+requests, ensures that clients can use request ordering to influence response
+order.
```

Comment:
How about moving the contents of this paragraph to the one above that talks about urgency, and going like: _When there are multiple responses with same urgency, a server SHOULD ..._

Reply:
I tried this and didn't like what I came up with. The progression from urgency, to incremental, to request order feels more natural to me. Combining the text as you suggest also produced a strange implication that request order is only important for requests at the same urgency, which I don't agree with.

I'd happily review a suggestion that avoids these problems. Perhaps we can make a separate editorial PR?

Reply:
:+1: I think I can work on a separate PR. That PR can point to this PR or master, depending on how we proceed.